frontend/src/features/admin-form/create/builder-and-design/MagicFormBuilder/components/MagicFormBuilderPromptModal.tsx

```diff
@@ -145,23 +154,28 @@ const TextPromptModalBodyContent = ({
           {...register('prompt', {
             required: 'Please enter a prompt',
             maxLength: {
-              value: MAX_CHAR,
-              message: 'Please enter at most 500 characters',
+              value: MFB_TEXT_PROMPT_MAX_CHAR,
+              message: `Please enter at most ${MFB_TEXT_PROMPT_MAX_CHAR} characters`,
             },
           })}
         />
         <FormErrorMessage>
-          {errors.prompt?.message && promptValue
-            ? `Your description cannot be longer than 500 characters. (${promptValue.length}/500)`
+          {isOverLimit
+            ? t('promptMaxLengthError', {
+                maxLength: MFB_TEXT_PROMPT_MAX_CHAR,
+                current: promptValue.length,
+              })
             : errors.prompt?.message}
         </FormErrorMessage>{' '}
-        {promptValue && !errors.prompt?.message && (
+        {promptValue && !errors.prompt?.message && !isOverLimit && (
           <FormHelperText
             color={
-              promptValue.length >= MAX_CHAR ? 'danger.500' : 'secondary.400'
+              promptValue.length >= MFB_TEXT_PROMPT_MAX_CHAR
```

Comment:
Instead of re-computing the error condition to determine the colour of the message, we should use [the existing expression](https://github.com/opengovsg/FormSG/pull/8965/files/583799a60ab8cf94c05af92bdd00cee0da871fe3..c440516f8484c77fe0944669dbc4e3907c399fb2#diff-dacd5fda04990f9055e9fd4c926a3568abeb0c092324a09d395d46afcbfcb877R142)!

```suggestion
              isOverLimit
```

Reply:
actually @kevin9foong had a better insight, I'll let him comment.